src/Microsoft.OData.Core/UriParser/Binders/BindingState.cs

```diff
@@ -151,6 +151,11 @@ internal List<CustomQueryOptionToken> QueryOptions
         /// </summary>
         internal bool InEntitySetAggregation { get; set; }
 
+        /// <summary>
+        /// The navigation source at the root.
+        /// </summary>
+        internal IEdmNavigationSource ResourcePathNavigationSource { get; set; }

```

Comment:
>ResourcePathNavigationSource [](start = 38, length = 28)

maybe name it as "RootNavigationSource"?

Reply:
`ResourcePath` is more appropriate. Check here https://docs.microsoft.com/en-us/odata/concepts/url-components

Reply:
The docstring says "The navigation source at the root" so `RootNavigationSource` sould also be a good fit if not better...

Reply:
Let me update the docstring. Should be `Navigation source at the resource path`